Makefile

```diff
@@ -141,27 +139,25 @@ endif
 
 ifeq ($(UNICORN_SHARED),yes)
 ifneq ($(filter MINGW%,$(UNAME_S)),)
-LIBRARY = $(BLDIR)/$(LIBNAME).$(EXT)
+LIBRARY = $(LIBNAME).$(EXT)
 else ifneq ($(filter CYGWIN%,$(UNAME_S)),)
-LIBRARY = $(BLDIR)/cyg$(LIBNAME).$(EXT)
```

Comment:
that "cyg" prefix was proposed by @iroiro123 (see https://github.com/unicorn-engine/unicorn/pull/323), who wanted that because "cyg" is a standard prefix for Cygwin DLL.

you are right that if we keep this, then yes Python must be changed to support this prefix. also, i am not really sure if the dependencies (supported DLLs) are the same, as with Msys.